components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceOrganizationServiceImpl.java

```diff
@@ -0,0 +1,250 @@
+        }
+    }
+
+    @GET
+    @Path("/devices")
+    @Override
+    public Response getDevicesInOrganization() {
+        List<DeviceOrganizationMetadataHolder> result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        try {
+            result = dops.getDevicesInOrganization();
+            if (result == null) {
+                String msg = "No devices exist in organization";
+                log.error(msg);
+                return Response.status(Response.Status.NOT_FOUND).entity(msg).build();
+            }
+            return Response.status(Response.Status.OK).entity(result).build();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error occurred while retrieving devices in organization";
+            log.error(msg, e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        } catch (InvalidConfigurationException e) {
+            log.error("failed to add operation", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+
+    @GET
+    @Path("/{parentId}/children")
+    @Override
+    public Response getChildrenByParentId(@PathParam("parentId") String parentId) {
+        if (parentId == null) {
+            String errorMessage = "The parameter of the parent ID is empty.";
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage).build();
+        }
+        List<DeviceOrganizationMetadataHolder> result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        try {
+            result = dops.getChildrenByParentId(parentId);
+            if (result == null) {
+                String msg = "No children connected to device";
+                log.error(msg);
+                return Response.status(Response.Status.NOT_FOUND).entity(msg).build();
+            }
+            return Response.status(Response.Status.OK).entity(result).build();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error occurred while retrieving children of device";
+            log.error(msg, e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        } catch (InvalidConfigurationException e) {
+            log.error("failed to add operation", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @GET
+    @Path("/nodes")
+    @Override
+    public Response generateNodes() {
+        List<DeviceOrganizationVisNode> result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        result = dops.generateNodes();
+        if (result.isEmpty()) {
+            String msg = "No Devices enrolled and visible";
+            log.error(msg);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
```

Comment:
IMO, this is not an error. We can log it as a debug log. Also response should be NOT_FOUND as there are no devices enrolled with the system in order to generate nodes.

Reply:
Fixed ce1b79e04ff7cb44e6a64e88ebab322946c55f47